aten/src/ATen/cpu/vec/functional_base.h

```diff
@@ -76,6 +76,30 @@ struct VecReduceAllSIMD<float, Op> {
   }
 };
 #endif // defined(CPU_CAPABILITY_AVX512)
+#if defined(CPU_CAPABILITY_NEON)
+template <typename Op>
+struct VecReduceAllSIMD<float, Op> {
+  static inline float apply(const Op& vec_fun, const Vectorized<float>& acc_vec) {
+    using Vec = Vectorized<float>;
+    Vec v = acc_vec;
+    // 128-bit shuffle: [a1, a2, a3, a4, a5, a6, a7, a8] -> [a5, a6, a7, a8, a1, a2, a3, a4]
+    Vec v1 = {v.get_high(), v.get_low()};
+    v = vec_fun(v, v1);
+    // 64-bit shuffle: [a5, a6, a7, a8, a1, a2, a3, a4] -> [a7, a8, a5, a6,  a3, a4, a1, a2]
+    float32x4_t v1_1 = vextq_f32(v.get_low(), v.get_low(), 2);
+    float32x4_t v1_2 = vextq_f32(v.get_high(), v.get_high(), 2);
+    v1 = {v1_1, v1_2};
+    v = vec_fun(v, v1);
+    // 32-bit shuffle: [a7, a8, a5, a6,  a3, a4, a1, a2] -> [a8, a7, a6, a5, a4, a3, a2, a1]
+    v1_1 = vrev64q_f32(v.get_low());
+    v1_2 = vrev64q_f32(v.get_high());
+    v1 = {v1_1, v1_2};
+    v = vec_fun(v, v1);
+
+    return v.get_low()[0];
+  }
+};
+#endif // defined(CPU_CAPABILITY_NEON)
```

Comment:
See [lines 38-56](https://github.com/Rohanjames1997/pytorch/pull/2/files#diff-0149700c2607377c8ea2741e6fcd9ee6adbba4ec49de6bb681375eb591932f60R38-R56) for reference AVX-2 code